compiler/evaluator.cc

```diff
@@ -25,35 +25,8 @@ namespace chainer_compiler {
 
 namespace {
 
-Dtype GetDtype(const chainerx::Array& a) {
-    switch (a.dtype()) {
-        case chainerx::Dtype::kBool:
-            return Dtype::kBool;
-        case chainerx::Dtype::kInt8:
-            return Dtype::kInt8;
-        case chainerx::Dtype::kInt16:
-            return Dtype::kInt16;
-        case chainerx::Dtype::kInt32:
-            return Dtype::kInt32;
-        case chainerx::Dtype::kInt64:
-            return Dtype::kInt64;
-        case chainerx::Dtype::kUInt8:
-            return Dtype::kUInt8;
-        case chainerx::Dtype::kFloat16:
-            return Dtype::kFloat16;
-        case chainerx::Dtype::kFloat32:
-            return Dtype::kFloat32;
-        case chainerx::Dtype::kFloat64:
-            return Dtype::kFloat64;
-    }
-    CHECK(false) << a;
-}
-
 Tensor* ArrayToTensor(const std::string& name, const chainerx::Array& a) {
-    Tensor::UniqueData data(std::malloc(a.GetNBytes()), &std::free);
-    memcpy(data.get(), a.Copy().ToNative().raw_data(), a.GetNBytes());
-    std::vector<int64_t> dims{a.shape().begin(), a.shape().end()};
-    return new Tensor(name, GetDtype(a), dims, data.release());
+    return new Tensor(name, a);
 }
 
 }  // namespace
```

Comment:
I think we can simplify tensor->array conversion thanks to the change for Tensor. https://github.com/pfnet-research/chainer-compiler/pull/228/files#diff-3bb5e0736d88411209b08e07e5eec2eeR77 could be just `state.SetArray(t->array());`. I think you need to add the new accessor function to `Tensor`. (or `Tensor::chx()` for consistency with Dtype?)